### PR TITLE
[et publish] Change canary base version to the current version

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Canaries need to be comparable to real versions so that we can gate certain CLI functionality to "greater than certain versions". One example is https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/projectUtils.ts#L102.

As discussed and recommended by @tsapeta in https://exponent-internal.slack.com/archives/C1QP38NQ5/p1705690294443739, we can just prefix canary versions with the current package version.

Closes ENG-11132.

# How

Change base prefix.

# Test Plan

Not tested yet. Will wait for approval before running a canary release. (need to do a canary release soon to start dogfooding new expo-updates features)

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — COMMENTED** [review]
Hmm, I think I meant a next version so the current one but bumped based on the changelog 🤔 Maybe the current one is fine, I'm open to discussions.

**Comment by wschurman** [comment]
> Hmm, I think I meant a next version so the current one but bumped based on the changelog 🤔 Maybe the current one is fine, I'm open to discussions.

Yeah, I thought about that too. I think that's fine as well, probably better too. I'll update this PR. Thanks for the review!

File: tools/src/publish-packages/tasks/publishCanary.ts [hunk 1]
```diff
@@ -66,8 +133,8 @@ export const publishCanaryPipeline = new Task<TaskArgs>(
       updateBundledNativeModulesFile,
       updateModuleTemplate,
       updateWorkspaceProjects,
-      packPackageToTarball,
-      publishPackages,
+      // packPackageToTarball,
+      // publishPackages,
```
Comment (wschurman) [inline tools/src/publish-packages/tasks/publishCanary.ts:137]: revert before landing if we do decide to go with this. (this was to test)

**Comment by wschurman** [comment]
Updated the PR to use the "next" version. The issue is that this now has the same issue as the normal publish flow in that it's not always predictable/consistent what the "next" version should be (since we don't always follow semver based on readmes, sometimes we only bump minor versions even if there has been a breaking change since the breaking change hasn't been published in a SDK). So we have to ask in the same way that we ask during the normal publish flow, which greatly reduces the usability of the command it seems.

A few solutions I see:
1. Always go by changelogs for publishing these and publishing normally. i.e. make it possible for a computer to do the bumping without user input.
2. Change the scheme somehow so that the next SDK's package versions can always be predicted (i.e. always bump major versions to match SDK major version for all packages)
3. Modify code in eas-cli (see link in summary) to just ignore versions with `canary` in the version.
4. Maybe we can just put the canaries on the current package version in the package.json (the last version) like this PR initially did. I'm not sure this is a good idea though since it may not actually the solve the problem for version comparisons.

I think I'm actually just leaning towards option 3 now. @tsapeta - what do you think?

**Comment by wschurman** [comment]
Option 3: https://github.com/expo/eas-cli/pull/2243

**Comment by tsapeta** [comment]
Yeah, option 3 sounds good enough for me. The current flow is not really a problem for the other packages and I agree implementing it well is a bit complex 😅 In the future, I would lean towards option 1 though.

**Pull request closed without merging**